internal/tcpchecker/logreporter.go

```diff
@@ -0,0 +1,38 @@
+package tcpchecker
+
+import "time"
+
+type LogReporter struct {
+	lastReport time.Time
+	lastOpen   bool
+	Log        interface {
+		Infof(msg string, args ...interface{})
+		Debugf(msg string, args ...interface{})
+	}
+}
+
+func (r *LogReporter) ReportPortCheck(report Report) {
```

Comment:
Could this normalization of the state be moved down a level in the abstraction? It seems to me that it is more close to the interface one would expect to be reported than to the reporter to understand. Does that even make sense?

So along `CheckPort`s reporting of type `Report` it would have a `Transition` or `Event` or something with values 
`closed`, `opened` etc.

```
Report{
	Address: address,
	Port:    port,
	Open:    false,
	Event:   "still open",
	Status:  "Scanner error",
	Error:   err,
	Content: output,
}
```

That would reduce the complexity of this switch and let et be reused between a `LogReporter` and `Prometheus` reporter.

It could be a method on `Report` as well if that makes more sense as you do some comparison to an early `Report` from how I read it.